### PR TITLE
[proxy/parser] Implement parser

### LINKED ISSUE #64: Implement parser for redis-proxy module
Should receive the same treatment as the worker-module in #55

### PR DESCRIPTION
**Pull request by cogitantium**
Also did reformat code with rustfmt, and add comments.

README.md is updated with help-message on proxy.

Resolves #64

### CODE REVIEW

File: proxy/src/main.rs [hunk 1]
```diff
@@ -1,65 +1,183 @@
...
+                .help("Specify the RabbitMQ exchange to connect to"),
+        )
+        .arg(
+            Arg::with_name("rabbitmq-routing-key")
+                .short("k")
+                .long("rmq-routing-key")
+                .env("SCRAPER_RABBITMQ_ROUTING_KEY")
+                .default_value("") // No routing-key by default
+                .value_name("KEY")
+                .help("Specify the RabbitMQ routing-key to connect to"),
```
Comment (NicEastvillage) [inline proxy/src/main.rs]: This argument will not used by the proxy.

Comment (cogitantium) [reply proxy/src/main.rs]: Fixed in 9559e10

File: proxy/src/main.rs [hunk 2]
```diff
@@ -1,65 +1,183 @@
...
+                .help("Specify the RabbitMQ-REDIS queue to connect to"),
+        )
+        .arg(
+            Arg::with_name("rabbitmq-exchange")
+                .short("e")
+                .long("rmq-exchange")
+                .env("SCRAPER_RABBITMQ_EXCHANGE")
+                .default_value("work")
+                .value_name("EXCHANGE")
+                .help("Specify the RabbitMQ exchange to connect to"),
```
[the remark this answers is not in the document]
Comment (cogitantium) [reply proxy/src/main.rs]: Fixed in 9559e10

File: proxy/src/main.rs [hunk 3]
```diff
@@ -1,65 +1,183 @@
...
+            Arg::with_name("rabbitmq-queue")
+                .short("q")
+                .long("rmq-queue")
+                .env("SCRAPER_RABBITMQ_QUEUE")
+                .default_value("frontier")
+                .value_name("QUEUE")
+                .help("Specify the RabbitMQ queue to connect to"),
+        )
+        .arg(
+            Arg::with_name("rabbitmq-redis-queue")
```
[the remark this answers is not in the document]
Comment (cogitantium) [reply proxy/src/main.rs]: Fixed in 9559e10

File: proxy/src/main.rs [hunk 4]
```diff
@@ -1,65 +1,183 @@
...
+                .help("Specify the redis set to connect to"),
+        )
+        .arg(
+            Arg::with_name("rabbitmq-queue")
+                .short("q")
+                .long("rmq-queue")
+                .env("SCRAPER_RABBITMQ_QUEUE")
+                .default_value("frontier")
+                .value_name("QUEUE")
+                .help("Specify the RabbitMQ queue to connect to"),
```
[the remark this answers is not in the document]
Comment (cogitantium) [reply proxy/src/main.rs]: Fixed in 9559e10

**Comment by cogitantium** [comment]
Also removes unneeded operations that the proxy previously did for testing.

**Pull request merged**